packages/dds/tree/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -530,24 +529,13 @@ function rebaseMark<TNodeChange>(
                         isSkipLikeReattach(newCurrMark),
                         0x4fe /* Only a skip-like reattach can overlap with a ReturnFrom */,
                     );
-                    if (
-                        newCurrMark.conflictsWith === baseMarkRevision ||
-                        (baseMark.type === "ReturnFrom" &&
-                            newCurrMark.conflictsWith === baseMark.detachedBy)
-                    ) {
-                        // The already populated cells that currMark aimed to reattach content into
-                        // are having their contents detached by baseMark.
-                        // This makes it possible for currMark to be active again.
-                        newCurrMark.detachedBy = baseMarkRevision;
-                        newCurrMark.detachIndex = baseInputOffset;
-                        delete (newCurrMark as CanConflict).conflictsWith;
-                        updateMoveSrcDetacher(moveEffects, newCurrMark.id, baseMarkRevision);
-                        updateMoveSrcPairing(
-                            moveEffects,
-                            newCurrMark.id,
-                            PairedMarkUpdate.Reactivated,
-                        );
-                    }
+                    // The already populated cells that currMark aimed to reattach content into
+                    // are having their contents detached by baseMark.
+                    // This makes it possible for currMark to be active again.
+                    newCurrMark.detachedBy = baseMarkRevision;
+                    newCurrMark.detachIndex = baseInputOffset;
+                    delete (newCurrMark as CanConflict).conflictsWith;
```

Comment:
Why is this cast necessary when we're this branch checks `newCurrMark.type === "ReturnTo"`?

Reply:
It's needed because the type of `newCurrMark` is inferred to be `ReturnTo & Conflicted & {
    lastDeletedBy?: undefined;
}` due to `isSkipLikeReattach`. The `Conflicted` part render the `conflictsWith` non-optional. I don't know if there's a way to have a type guard that tells the compiler "the field is populated but is allowed not to be".